concrete-test/src/raw/statistical_test.rs

```diff
@@ -0,0 +1,85 @@
+//! A module containing statistical testing entry points for raw integers
+use crate::raw::generation::RawUnsignedIntegers;
+use concrete_commons::dispersion::{DispersionParameter, Variance};
+use concrete_core::backends::core::private::math::random::RandomGenerator;
+use kolmogorov_smirnov;
+
+/// A function performing a kolmogorov smirnov statistical test.
+///
+/// The `tested` argument points to an array of samples which is tested for normality. The
+/// `expected_means` argument encodes the mean expected for each element of the `tested` array (can
+/// be different for each element). The `expected_variance` argument encodes the variance expected
+/// for the whole sample array.
+#[allow(dead_code)]
+pub fn assert_noise_distribution<Raw>(
+    tested: &[Raw],
+    expected_means: &[Raw],
+    expected_variance: Variance,
+) where
+    Raw: RawUnsignedIntegers,
+{
+    let std_dev = expected_variance.get_standard_dev();
+    let confidence = 0.95;
```

Comment:
If I understand this parameter right, then this means that we have a 5% chance of the test rejecting the null hypothesis even though the samples are in fact from the same distribution. Is that right? If so, the 0.95 seems low, because at 10 repetitions (see `crate::REPETITIONS`), there is a >40% probability that at least one of them will fail even if the distributions are the same. This seems like a lot, especially if we want to use this for all tests. But I might be misinterpreting the parameter and what it actually means.

Reply:
Good catch ! 

Which values do you think would make sense for `crate::REPETITIONS` and `confidence` ?

Reply:
Actually, I think I made a mistake here. After looking a bit closer at it, I think a confidence of 0.95 means that the probability of the distributions being the same is less than 5% if the test rejects. That's different from saying that the test will reject with at most 5% probability if the two distributions are the same. Classic Bayesian mistake. I'm not sure how to compute one from the other at this point (what we are interested in here is usually called the significance level, or type 1 error), but after what we've talked about today it looks like there are other things to worry about first.

Reply:
Ok I leave it that way for now, and we will rework this later on then.